samples/cpp/tutorial_code/ImgTrans/imageSegmentation.cpp

```diff
@@ -0,0 +1,168 @@
+    for( int x = 0; x < src.rows; x++ ) {
+      for( int y = 0; y < src.cols; y++ ) {
+          if ( src.at<Vec3b>(x, y) == Vec3b(255,255,255) ) {
+            src.at<Vec3b>(x, y)[0] = 0;
+            src.at<Vec3b>(x, y)[1] = 0;
+            src.at<Vec3b>(x, y)[2] = 0;
+          }
+        }
+    }
+
+    // Show output image
+    imshow("Black Background Image", src);
+//! [black_bg]
+
+//! [sharp]
+    // Create a kernel that we will use for accuting/sharpening our image
+    Mat kernel = (Mat_<float>(3,3) <<
+            1,  1, 1,
+            1, -8, 1,
+            1,  1, 1); // an approximation of second derivative, a quite strong kernel
+
+    // do the laplacian filtering as it is
+    // well, we need to convert everything in something more deeper then CV_8U
+    // because the kernel has some negative values,
+    // and we can expect in general to have a Laplacian image with negative values
+    // BUT a 8bits unsigned int (the one we are working with) can contain values from 0 to 255
+    // so the possible negative number will be truncated
+    Mat imgLaplacian;
+    Mat sharp = src; // copy source image to another temporary one
+    filter2D(sharp, imgLaplacian, CV_32F, kernel);
+    src.convertTo(sharp, CV_32F);
+    Mat imgResult = sharp - imgLaplacian;
+
+    // convert back to 8bits gray scale
+    imgResult.convertTo(imgResult, CV_8UC3);
+    imgLaplacian.convertTo(imgLaplacian, CV_8UC3);
+
+    // imshow( "Laplace Filtered Image", imgLaplacian );
+    imshow( "New Sharped Image", imgResult );
+//! [sharp]
+
+    src = imgResult; // copy back
+
+//! [bin]
+    // Create binary image from source image
+    Mat bw;
+    cvtColor(src, bw, CV_BGR2GRAY);
+    threshold(bw, bw, 40, 255, CV_THRESH_BINARY | CV_THRESH_OTSU);
+    imshow("Binary Image", bw);
+//! [bin]
+
+//! [dist]
+    // Perform the distance transform algorithm
+    Mat dist;
+    distanceTransform(bw, dist, CV_DIST_L2, 3);
+
+    // Normalize the distance image for range = {0.0, 1.0}
+    // so we can visualize and threshold it
+    normalize(dist, dist, 0, 1., NORM_MINMAX);
+    imshow("Distance Transform Image", dist);
+//! [dist]
+
+//! [peaks]
+    // Threshold to obtain the peaks
+    // This will be the markers for the foreground objects
+    threshold(dist, dist, .4, 1., CV_THRESH_BINARY);
+
+    // Dilate a bit the dist image
+    Mat kernel1 = Mat::ones(3, 3, CV_8UC1);
+    dilate(dist, dist, kernel1);
+    imshow("Peaks", dist);
+//! [peaks]
+
+//! [seeds]
+    // Create the CV_8U version of the distance image
+    // It is needed for findContours()
+    Mat dist_8u;
+    dist.convertTo(dist_8u, CV_8U);
+
+    // Find total markers
+    vector<vector<Point> > contours;
+    findContours(dist_8u, contours, CV_RETR_EXTERNAL, CV_CHAIN_APPROX_SIMPLE);
+
+    // Create the marker image for the watershed algorithm
+    Mat markers = Mat::zeros(dist.size(), CV_32SC1);
+
+    // Draw the foreground markers
+    for (int i = 0; i < contours.size(); i++)
```

Comment:
iterators on a vector are usually done by a size_t element in order to remove warnings. Just replace `for (int i = 0; i < contours.size(); i++)` by Just replace `for (size_t i = 0; i < contours.size(); i++)`